pkg/ccl/changefeedccl/sink_cloudstorage.go

```diff
@@ -308,8 +309,19 @@ type cloudStorageSink struct {
 
 	asyncFlushActive bool
 	flushCtx         context.Context
-	flushGroup       sync.WaitGroup
-	flushErr         atomic.Value
+	flushQueue       struct {
+		syncutil.Mutex
+		err           error
+		q             []destFile
+		numGoRoutines int
+	}
+	// Signaled whenever changes are made to flushQueue struct.
+	flushC *sync.Cond
```

Comment:
nit: put this in the `flushQueue` and don't make it a pointer.

Reply:
ack. done.